src/public/app/widgets/buttons/attachments_actions.ts

```diff
@@ -92,20 +99,21 @@ export default class AttachmentActionsWidget extends BasicWidget {
 
     doRender() {
         this.$widget = $(TPL);
-        this.dropdown = Dropdown.getOrCreateInstance(this.$widget.find("[data-bs-toggle='dropdown']"));
+        this.dropdown = Dropdown.getOrCreateInstance(this.$widget.find("[data-bs-toggle='dropdown']")[0]);
         this.$widget.on("click", ".dropdown-item", () => this.dropdown.toggle());
 
         this.$uploadNewRevisionInput = this.$widget.find(".attachment-upload-new-revision-input");
         this.$uploadNewRevisionInput.on("change", async () => {
-            const fileToUpload = this.$uploadNewRevisionInput[0].files[0]; // copy to allow reset below
-            this.$uploadNewRevisionInput.val("");
-
-            const result = await server.upload(`attachments/${this.attachmentId}/file`, fileToUpload);
 
-            if (result.uploaded) {
-                toastService.showMessage(t("attachments_actions.upload_success"));
-            } else {
-                toastService.showError(t("attachments_actions.upload_failed"));
+            const fileToUpload = this.$uploadNewRevisionInput[0].files?.item(0);  // copy to allow reset below
+            this.$uploadNewRevisionInput.val("");
+            if (fileToUpload) {
```

Comment:
Use a guard expression instead where you return if not found, rather than increasing indentation for the main code.

Reply:
I wanted to do a small refactor here anyways (to break out out some of the code in `doRender`  into their own smaller methods) – and have done exactly that in my local branch already – if it is OK, I'll keep it like this in this PR and will introduce your requested change in my follow up PR.